### ISSUE TITLE
Why do some appendTo methods not return? They exit only via an IllegalArgumentExpection.

### ISSUE THREAD

**Issue by yousifalyousifi** [virtuald/curvesapi#7]
CardinalSpline, CatmullRomSpline, and LagrangeCurve objects have appendTo() methods that exit only via an IllegalArgumentException.

For example, from CardinalSpline:
```
while (true) {
	int index_i = gi.index_i();
	int count_j = gi.count_j();

	for (int i = 0; i < 4; i++) {
		if (!gi.hasNext())
			throw new IllegalArgumentException("Group iterator ended early");
		sharedData.pt[i] = cp.getPoint(gi.next()).getLocation();
	}

	gi.set(index_i, count_j);
	gi.next();

	BinaryCurveApproximationAlgorithm.genPts(this, 0.0, 1.0, mp);
}
```

I compared this with the source from the older project on SourceForge and in there they ```return``` instead of throwing the exception.

Basically if the Curve object is any of these three, then I have to make sure appendTo() is in a try/catch as there is no way it won't throw the IAE.

I know this is a pretty late question :), but do you remember why you made this change?

Thanks

**Comment by virtuald** [comment]
I think I ran into a case where I was using the wrong type of Multipath on some particular curve type and it wasn't telling me that an error occurred, so I went and changed all of the curves with the same pattern without looking too closely at them. Of course, there aren't any tests for this library, so clearly I made a mistake.

I don't have time at the moment to fix it, but I would accept a pull request with a fix.

**Comment by yousifalyousifi** [comment]
Thanks for the speedy reply!

I am comparing the old project and this one to see what the changes are and I notice that in the old project they just ```return``` on invalid input (they return quietly as they put it), they don't throw any exceptions.

So I think as you were replacing these quiet returns with exceptions, you saw a normal return and replaced it with an exception throw.

I'll submit a pull request with that simple change. I hope there aren't wider implications to the change. I found this library an hour ago, and just trying to understand basic usage. Your test example helped and the old project's demo helped a bit too.

I am currently using it within [Processing](https://processing.org/) and so far so good.

**Comment by virtuald** [comment]
The main reason I forked the library into maven was I needed the library for use with POI to render some NURBS splines... you can look in the POI source code for some example usages.

* https://github.com/apache/poi/blob/afc30856848e91b537c788723ca778d7c3890cb4/src/ooxml/java/org/apache/poi/xdgf/usermodel/section/geometry/NURBSTo.java
* https://github.com/apache/poi/blob/a98eaeb917943adcd6bfe04c9e446d59a94f7de2/src/ooxml/java/org/apache/poi/xdgf/geom/SplineRenderer.java

**Comment by virtuald** [comment]
Fixed in #8
